src/server/routes/api/classrooms/index.js

```diff
@@ -293,23 +338,44 @@ router.post("/:classroomId/answer", async (req, res) => {
     });
   }
 
-  const question = classroom.questions.filter(q => q._id.toString() === _id)[0];
+  const question = classroom.questions.filter(q => q._id.toString() === questionId)[0];
 
   if (!question) {
     return res.status(404).json({
       success: false,
-      message: `that classroom does not have a question with _id ${_id}`,
+      message: `That classroom does not have a question with _id ${_id}`,
     });
   }
 
-  question.answers.push({
+  const answerDoc = {
     user: req.user._id,
     answer,
     ts: Date.now(),
+  };
+
+  await Classrooms.findOneAndUpdate({
+    _id: req.params.classroomId,
+    questions: {
+      $elemMatch: { _id: question._id, },
+    }, 
+  }, {
+    $addToSet: { 'questions.$.answers': answerDoc, }
   });
 
-  await Classrooms.findByIdAndUpdate(req.params.classroomId, {
-    questions: classroom.questions,
+  const updatedC = await Classrooms.findById(req.params.classroomId);
+  const updatedQ = updatedC.questions.filter( q => q._id.toString() === questionId)[0];
+  console.log("Updated question:", updatedQ);
+  if(updatedQ.answers.length < 1)
+    return res.status(300).send({ success: false, message: "Answer not created successfully" });
+  const newAnswerDoc = updatedQ.answers[updatedQ.answers.length-1];
+
+  // update users with new answer over sockets
+  req.app.ee.emit("new_answer", {
+    classroom: req.params.classroomId,
+    questionId: updatedQ._id.toString(),
+    answerId: newAnswerDoc._id.toString(),
```

Comment:
same race as before, and see above comment on the emit. i think the same logic applies here